plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -299,15 +299,27 @@ public void dropSchema(ConnectorSession session, String schemaName)
     @Override
     protected void dropTable(ConnectorSession session, RemoteTableName remoteTableName, boolean temporaryTable)
     {
-        String sql = "DROP TABLE " + quoted(remoteTableName);
+        String quotedTable = quoted(remoteTableName);
+        // By default, when dropping a table, oracle does not wait for the table lock.
+        // If another transaction is using the table at the same time, DROP TABLE will throw.
```

Comment:
If it had thrown we would observe those failures on the CI. Not only observe unexpected remainder tables after tests, right?

Reply:
Can we have a test which reproduces original issue and showing that fix fixes it?

Reply:
@losipiuk yes, that is indeed what happened. see my comment [here](https://github.com/trinodb/trino/issues/16277#issuecomment-1555182734)

Reply:
Thanks. ONe more question. Is this locking issue specific to `PURGE` flag. That is should we apply the lock protocol only to temporary tables?

Reply:
I don't believe so - the locking issue is an oddity of oracle. When mysql or postgres drop a table, they both lock it, and will block while waiting for that lock. 

I suppose there's an argument that could be made that, if you're dropping an oracle table through trino, you'd expect it to act a typical Oracle table, and so you might expect it to throw if there is another active transaction on it, but IMO this is really bizarre behavior, and it'd only count as a non-user-caused query failure in our reporting. This is getting philosophical.

Ultimately, I think it makes more sense for a drop table to always succeed, and I don't think doing so will cause any issues. That being said, I'm open to only locking temporaries if that's what you'd prefer.

Reply:
Isn't it possible that locks on tables not owned by Trino may never be acquired (or take a very long time)? Is there some reasonable timeout?

I think the lock should only be taken against tables we are sure about the ownership of - temporary ones.

Reply:
> @hashhar please make a call

Oh - I see you already suggested what should be done :)